src/storage/storage.h

```diff
@@ -241,6 +241,12 @@ typedef struct StorageRemoveParam
 
 void storageRemove(const Storage *this, const String *fileExp, StorageRemoveParam param);
 
+// Create a hard or soft link
```

Comment:
I'm not sure that linkType deserves a required parameter since it we use symlink in all but one place. We'd need a StorageNewWriteParam struct to contain the parameter. We could either keep the enum with SymLink as the default or just use a boolean `hard`. I'm fine with either.

If we keep the enum we'll want to use symlink instead of softlink. Or change the current instances of "sym" to "soft" but not sure that's worth it. Either way let's keep the naming consistent.

Reply:
Followed the model of the other functions and created StorageInterfaceLinkCreateParam to contain the parameter. Kept enums. Changed soft to sym to stay with original naming and keep things consistent.